packages/keystone/KSv5v6/v5/keystone.js

```diff
@@ -1,12 +1,34 @@
 const { Keystone: DefaultKeystone } = require('@open-keystone/keystone')
+const { GraphQLScalarType, GraphQLError } = require('graphql')
+const Upload = require('graphql-upload/Upload.js')
 
 const { _patchResolverWithGQLContext } = require('../utils/resolvers')
 
+
+const FileMetaScalar = new GraphQLScalarType({
```

Comment:
It's not clear. One of the main reasons to move the file upload logic to one URL is to restrict GQL style upload. Probably you are trying to add backward compatibility for existing uploads. But it's not clear for me.

Reply:
Can we add an environment variable to disable the legacy backward‑compatibility logic? Like FILE_COMPATIBILITY_MODE=1

Reply:
> It's not clear. One of the main reasons to move the file upload logic to one URL is to restrict GQL style upload. Probably you are trying to add backward compatibility for existing uploads. But it's not clear for me.

To be honest I don't understand the question. You just answered it yourself.
I don't see any other options for solving this problem - using Scalar looks neat and not bulky

Reply:
> Can we add an environment variable to disable the legacy backward‑compatibility logic? Like FILE_COMPATIBILITY_MODE=1

I can do it, but it seems like we'll shoot ourselves in the foot during the final migration